tests/e2e/mock_llm_trace.py

```
@@ -0,0 +1,643 @@
+        "source": source,
+        "responses": responses,
+        "next_response": 0,
+        "expected_user_inputs": expected_user_inputs,
+        "request_hints": request_hints,
+        "pending_tool_calls": [],
+        "tool_call_id_aliases": {},
+        "error": None,
+    }
+
+
+def _next_llm_trace_response(
```

Ruff (0.16.0)

[warning] Too many branches (14 > 12)

(PLR0912)